features/camel/src/main/feature/feature.xml

```diff
@@ -16,285 +16,561 @@
           xmlns="http://karaf.apache.org/xmlns/features/v1.3.0"
           xsi:schemaLocation="http://karaf.apache.org/xmlns/features/v1.3.0 http://karaf.apache.org/xmlns/features/v1.3.0">
 
-    <repository>mvn:org.apache.karaf.features/spring/${karaf.version}/xml/features</repository>
-    <repository>mvn:ddf.features/cxf/${project.version}/xml/features</repository>
-    <repository>mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features</repository>
+  <repository>mvn:ddf.features/cxf/${project.version}/xml/features</repository>
+  <repository>mvn:org.apache.karaf.features/spring/${karaf.version}/xml/features</repository>
+  <repository>mvn:org.apache.karaf.features/enterprise/${karaf.version}/xml/features</repository>
 
-    <!--START Camel Features-->
-    <!-- DO NOT EDIT
-        All feature definitions come from 3rd party feature file:
-        <repository>mvn:org.apache.camel.karaf/apache-camel/2.24.2/xml/features</repository>
+  <!--START Camel Features-->
+  <!-- DO NOT EDIT
+      All feature definitions come from 3rd party feature file:
+      <repository>mvn:org.apache.camel.karaf/apache-camel/3.4.4/xml/features</repository>
 
-        All version properties for these Camel features are defined in the platform-app pom.
+      All version properties for these Camel features are defined in the platform-app pom.
 
-        DDF Changes made:
-        - removed resolver="(obr)" from every feature
-        - added  to every feature
-        - replaced ${project.version} with ${camel.version}
-        - commented out xml-specs-api feature and replaced xml-specs-api with cxf-specs in camel-core feature
-        - added camel as feature prerequisite in camel-http to keep things ordered
-        - added cxf as feature prerequisite in camel to keep things ordered
-        - removed features that are not needed
-        - updated versions of qpid and proton versions and added netty bundles in camel-amqp
-        - replaced bundles in camel-jms
-        - replaced cxf-http-provider with cxf-http-jetty
-        - the versions for proton-j and qpid-jms-client used in camel-amqp need to be kept in sync with the versions used in artemis-amqp
-        - updated proton, qpid-jms, and netty versions to stay in sync with Camel 2.24.2 dependencies
+      DDF Changes made:
+      - replaced ${project.version} with ${camel.version}
+      - commented out xml-specs-api feature and replaced xml-specs-api with cxf-specs in camel-core feature
```

Comment:
Rather than copying these features and having to keep them in sync with camel can we just use org.apache.features.xml featureReplacements?

Reply:
If we do that, then when we add features to the repository at compile time, we'd have to add bundles for both the old feature and our replacement, I think. If the reason for replacing the feature is to update vulnerable dependencies, then we have a problem because we're still packaging the vulnerable bundles in the distro, where they could be used (i.e. if somebody accidentally removes the replacement in `org.apache.features.xml`. Plus we'd get CVEs for the distro in OWASP reports.